minecraft/realms/realms.go

```diff
@@ -0,0 +1,213 @@
+	Accepted   bool   `json:"accepted"`
+	Online     bool   `json:"online"`
+	Permission string `json:"permission"`
+}
+
+// Realm returned from the api
+type Realm struct {
+	// the unique id for this realm.
+	Id int `json:"id"`
+	// The subscription ID of the realm.
+	RemoteSubscriptionId string `json:"remoteSubscriptionId"`
+	// This is always an empty string.
+	Owner string `json:"owner"`
+	// The XboxUserID (XUID) of the owner.
+	OwnerUUID string `json:"ownerUUID"`
+	// The name of the Realm.
+	Name string `json:"name"`
+	// This is always an empty string.
+	Motd string `json:"motd"`
+	// The default permission level of the Realm world.
+	// one of ["MEMBER", "OPERATOR"]
+	DefaultPermission string `json:"defaultPermission"`
+	// The current state of the realm
+	// one of: ["OPEN", "CLOSED"]
+	State string `json:"state"`
+	//  The days remaining before renewal of the Realm as an integer.
+	// (always 0 for Realms where the current user is not the owner)
+	DaysLeft int `json:"daysLeft"`
+	// whether the Realm has expired as a trial or not.
+	Expired bool `json:"expired"`
+	// whether the Realm has expired as a trial or not.
+	ExpiredTrial bool `json:"expiredTrial"`
+	// whether the Realm is in its grace period after expiry or not.
+	GracePeriod bool `json:"gracePeriod"`
+	// The world type of the currently loaded world
+	// one of: ["NORMAL", "FLAT"?]
+	WorldType string `json:"worldType"`
+	// Players is a list of the players currently online in the realm
+	// NOTE: this is only sent when directly requesting a realm
+	Players []RealmPlayer `json:"players"`
+	// MaxPlayers how many player slots this realm has
+	MaxPlayers int `json:"maxPlayers"`
+	// always null
+	MinigameName string `json:"minigameName"`
+	// always null
+	MinigameId string `json:"minigameId"`
+	// always null
+	MinigameImage string `json:"minigameImage"`
+	// unused, always 1
+	ActiveSlot int `json:"activeSlot"`
+	// unused, always null
+	Slots []struct{} `json:"slots"`
+	// Unknown, always false. (even when member or owner)
+	Member bool `json:"member"`
+	// The ID of the associated Xbox Live club as an integer.
+	ClubId int `json:"clubId"`
+	// Unknown, always null
+	SubscriptionRefreshStatus struct{} `json:"subscriptionRefreshStatus"`
+
+	// instance of RealmsApi that this belongs to
+	_realmsApi *RealmsApi
+}
+
+// GetAddress requests the address and port to connect to this realm from the api
+// if wait is true it will poll the api until the server is started
+// else it will just return the http error
+func (r *Realm) GetAddress(ctx context.Context, wait bool) (address string, err error) {
+	ticker := time.NewTicker(time.Second * 3)
+	defer ticker.Stop()
+	i := 0
+	for range ticker.C {
+		i++
+		body, status, err := r._realmsApi.getRequest(ctx, fmt.Sprintf("/worlds/%d/join", r.Id))
+		if err != nil {
+			if status == 503 && wait {
+				fmt.Printf("Waiting for the realm to start... %d\033[K\r", i)
+				continue
+			}
+			return "", err
+		}
+		println()
```

Comment:
Would remove this.

Reply:
it needs a newline there because of the ansi escape above,
should i remove that too?